src/project-manager.ts

```diff
@@ -26,11 +28,14 @@ export class ProjectManager implements Disposable {
 	private rootPath: string;
 
 	/**
-	 * Workspace subtree (folder) -> JS/TS configuration mapping.
+	 * (Workspace subtree (folder) -> TS or JS configuration) mapping.
 	 * Configuration settings for a source file A are located in the closest parent folder of A.
 	 * Map keys are relative (to workspace root) paths
 	 */
-	private configs: Map<string, ProjectConfiguration>;
+	private configs = {
+		js: new Map<string, ProjectConfiguration>(),
+		ts: new Map<string, ProjectConfiguration>()
```

Comment:
I just thought, why do we wanna keep different maps at all? The only reason they conflict is because we decided to map from directory name to ProjectConfiguration. Why not simply map from jsconfig.json/tsconfig.json path (or URI) to ProjectConfiguration?